### PR TITLE
Database api remake

### PR DESCRIPTION
**Pull request by bagrorg**

### CODE REVIEW

File: db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt [hunk 1]
```diff
@@ -21,100 +21,80 @@ class BotsDatabase {
...
-            exec("select create_double_parameter('$paramName', '$paramDescription', $defaultParamValue);") {
-                it.next()
-                it.getInt(1)
-            }
+            exec("select set_double_parameter($botId, $paramId, $newValue);")
         }
     }
 
-    fun addIntParameter(paramId: Int, stratId: Int): Int? {
+    fun setStringParameter(botId: Int, paramId: Int, newValue: String) {
```
Comment (SBOne-Kenobi) [inline db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:45]: Is it really necessary?

File: db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt [hunk 2]
```diff
@@ -21,100 +21,80 @@ class BotsDatabase {
...
+            exec("select get_double_parameter($botId, $paramId);") {
                 it.next()
-                it.getInt(1)
+                it.getDouble(1)
             }
         }
     }
 
-    fun getDoubleParameter(paramId: Int, botId: Int): Double? {
+    fun getStringParameter(botId: Int, paramId: Int): String? {
```
Comment (SBOne-Kenobi) [inline db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:78]: Same

File: db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt [hunk 3]
```diff
@@ -128,10 +108,11 @@ class BotsDatabase {
                     ops.add(
```
Comment (SBOne-Kenobi) [inline db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:108]: getOperations fails on empty result, add more checks (such as in getBotsByStrategy) in this and other methods

File: db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt (hunk 1 above)
Comment (bagrorg) [reply db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:45]: You mean `string` parameter?

File: db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt (hunk 2 above)
Comment (bagrorg) [reply db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:78]: Same

File: db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt (hunk 3 above)
Comment (bagrorg) [reply db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:108]: Yeap, this is known issue, will fix this

File: db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt (hunk 1 above)
Comment (SBOne-Kenobi) [reply db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:45]: Yes

Comment (bagrorg) [reply db-wrapper/src/main/kotlin/backend/db/bots/BotsDatabase.kt:45]: @sergeypospelov requested this

**Review by sergeypospelov — APPROVED** [review]

**Pull request merged**